### PR TITLE
Bug 1553800 - Re-organize FFI component into better groups of code

### PR DESCRIPTION
**Pull request by badboy**
This is the first step towards a more nicely organized FFI component (both on the Rust and the Kotlin side)

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

**Comment by badboy** [comment]
No idea why it failed. Restarting.

File: glean-core/ffi/src/boolean.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+use std::convert::TryFrom;
```
Comment (Dexterp37) [inline glean-core/ffi/src/boolean.rs]: This is missing the license header.

File: glean-core/ffi/src/counter.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
+use std::convert::TryFrom;
```
Comment (Dexterp37) [inline glean-core/ffi/src/counter.rs]: This is missing the license header.

File: glean-core/ffi/src/datetime.rs [hunk 3]
```diff
@@ -0,0 +1,122 @@
+use std::convert::TryFrom;
```
Comment (Dexterp37) [inline glean-core/ffi/src/datetime.rs]: This is missing the license header.

File: glean-core/ffi/src/ping_type.rs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+use ffi_support::{define_handle_map_deleter, ConcurrentHandleMap, FfiStr};
```
Comment (Dexterp37) [inline glean-core/ffi/src/ping_type.rs]: This is missing the license header.

File: glean-core/ffi/src/string.rs [hunk 5]
```diff
@@ -0,0 +1,85 @@
+use std::convert::TryFrom;
```
Comment (Dexterp37) [inline glean-core/ffi/src/string.rs]: This is missing the license header.

File: glean-core/ffi/src/string_list.rs [hunk 6]
```diff
@@ -0,0 +1,104 @@
+use std::convert::TryFrom;
```
Comment (Dexterp37) [inline glean-core/ffi/src/string_list.rs]: This is missing the license header.

**Pull request merged**